File: app/src/main/java/com/example/biblio/fragments/ProfileFragment.java
package com.example.biblio.fragments;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.example.biblio.LoginActivity;
import com.example.biblio.databinding.ProfileFragmentBinding;
import com.example.biblio.helpers.XFragment;

import static android.app.Activity.RESULT_OK;


public class ProfileFragment extends XFragment {
    private static final int RC_SIGN_IN = 1;
    private ProfileFragmentBinding binding;

    public ProfileFragment() {
        super(ProfileFragment.class);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = ProfileFragmentBinding.inflate(inflater, container, false);
        setUpButtons();

        binding.emailLoginBtn.setOnClickListener(view -> {
            Intent i = new Intent(getActivity(), LoginActivity.class);
            startActivityForResult(i, RC_SIGN_IN);
        });

        binding.settingsBtn.setOnClickListener(view -> moveTo(new SettingsFragment(), SettingsFragment.TAG));
        return binding.getRoot();
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == RC_SIGN_IN && resultCode == RESULT_OK)
            replaceWith(new LoggedProfileFragment());
    }

    private void setUpButtons() {
        binding.infoTv.setVisibility(View.VISIBLE);
        binding.emailLoginBtn.setVisibility(View.VISIBLE);
    }
}

File: app/src/main/java/com/example/biblio/fragments/ProfileFragment.kt
package com.example.biblio.fragments

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.biblio.LoginActivity
import com.example.biblio.SettingsActivity
import com.example.biblio.databinding.FragmentProfileBinding
import com.example.biblio.helpers.XFragment

class ProfileFragment : XFragment(ProfileFragment::class.java) {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val binding = FragmentProfileBinding.inflate(inflater, container, false)
        binding.emailLoginBtn.setOnClickListener {
            val intent = Intent(activity, LoginActivity::class.java)
            startActivityForResult(intent, RC_SIGN_IN)
        }
        binding.settingsBtn.setOnClickListener { startActivity(Intent(activity, SettingsActivity::class.java)) }
        return binding.root
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == RC_SIGN_IN && resultCode == Activity.RESULT_OK) replaceWith(LoggedProfileFragment())
    }

    companion object {
        private const val RC_SIGN_IN = 1
    }
}